lib/active_merchant/billing/gateways/deepstack.rb

```diff
@@ -0,0 +1,388 @@
+module ActiveMerchant #:nodoc:
+  module Billing #:nodoc:
+    class DeepstackGateway < Gateway
+      self.test_url = 'https://api.sandbox.deepstack.io'
+      self.live_url = 'https://api.deepstack.io'
+
+      self.supported_countries = ['US']
+      self.default_currency = 'USD'
+      self.supported_cardtypes = %i[visa master american_express discover]
+      self.money_format = :cents
+
+      self.homepage_url = 'https://deepstack.io/'
+      self.display_name = 'Deepstack Gateway'
+
+      STANDARD_ERROR_CODE_MAPPING = {}
+
+      def initialize(options = {})
+        requires!(options, :publishable_api_key, :app_id, :shared_secret)
+        @publishable_api_key, @app_id, @shared_secret = options.values_at(:publishable_api_key, :app_id, :shared_secret)
+        super
+      end
+
+      # Charge with capture set to true
+      def purchase(money, payment, options = {})
+        post = {}
+        add_payment(post, payment, options)
+        add_order(post, money, options)
+        # Set this to true to capture
+        post[:transaction][:capture] = true
```

Comment:
Related to the other comment, setting `post[:transaction][:capture] = true` here assumes that `add_order` will always be called before this line, otherwise `post[:transaction]` would not yet be initialized. This could lead to unintentional bugs in the future. There's an opportunity to use the `||=` operator here as well, and/or revisit how you're conditionally setting `post[:transaction][:capture]` for purchases via another method.

Reply:
I have created a new method for setting this value to true that checks if post[:transaction] exists first